clevertap-core/src/main/java/com/clevertap/android/sdk/store/preference/CTPreference.kt

```diff
@@ -0,0 +1,196 @@
+package com.clevertap.android.sdk.store.preference
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.SharedPreferences
+import android.content.SharedPreferences.Editor
+import androidx.annotation.RestrictTo
+import androidx.annotation.RestrictTo.Scope.LIBRARY_GROUP
+import androidx.annotation.VisibleForTesting
+import androidx.annotation.WorkerThread
+import java.lang.ref.WeakReference
+
+/**
+ * The `CTPreference` class implements the `ICTPreference` interface and provides methods
+ * to read and write various types of data to shared preferences. It ensures safe handling
+ * of shared preferences operations in a weakly referenced context to prevent memory leaks.
+ *
+ * @param context The Android application context.
+ * @param prefName The name of the shared preferences file.
+ */
+@RestrictTo(LIBRARY_GROUP)
+class CTPreference(context: Context, private var prefName: String? = null) : ICTPreference {
```

Comment:
rather than having a null check for ctpref and returning default, we should make sure in higher layer the pref is correctly initialised and handle it there itself.

Reply:
lets do it if it is feasible or we can add to debt items

Reply:
tech debt